layer.go

```diff
@@ -67,46 +69,49 @@ type Layer struct {
 	// finalHandler stores the final middleware chain handler.
 	finalHandler http.Handler
 
-	// memo stores the memoized middleware call chain by specific phase.
-	memo map[string]http.Handler
+	// parent stores the parent middleware layer to use. Use SetParent(parent).
+	parent Middleware
 
-	// stack stores the plugins registered in the current middleware instance.
+	// Pool stores the phase-specific middleware handlers stack.
 	Pool Pool
 }
 
 // New creates a new middleware layer.
 func New() *Layer {
-	return &Layer{Pool: make(Pool), memo: make(map[string]http.Handler), finalHandler: FinalHandler}
+	return &Layer{Pool: make(Pool), finalHandler: FinalHandler}
 }
 
-// Flush flushes the plugins stack.
+// Flush flushes the middleware pool.
 func (s *Layer) Flush() {
-	s.Pool = Pool{}
+	s.Pool = make(Pool)
```

Comment:
One word: nothing.
They're syntax aliases. 

In Go you can do it in two ways, with or without operator. The relations are: 
- `make` initializes a built-in primitive types, such as `maps`, `slices` and `channels`.
- `new` initializes a given struct and returns its pointer.

There're some other benefits of using `make` with slices, like defining pre-fixed length.

Syntactically speaking, these expressions are aliases:
`map[string]string{}` == `make(map[string]string)`
`&Struct{}` == `new(Struct)`

It's more idiomatic use `make` in general, but adds some boilerplace when constructing with default members. `new` for instance is commonly less idiomatic (but perfectly valid).